index.rst

```diff
@@ -1,59 +1,367 @@
 
-   Run: ``make html`` and ``open _build/html/index.html`` to preview your work.
-   See the README at https://github.com/lsst-sqre/lsst-technote-bootstrap or
-   this repo's README for more info.
+There are several types of collections, which are quite different in how they how they are written, but almost identical to readers.
+A full description of the types of collections can be found in the `daf_butler documentation`_, but a brief summary here should be sufficient for the proposals in this document:
 
-   Feel free to delete this instructional comment.
+- A ``RUN`` collection (or just "a run") is the only type of collection that is intrinsic to a dataset - a dataset is added to ``RUN`` collection when it is first added to a repository, and remains in that ``RUN`` collection until/unless it is deleted entirely.
+- A ``CHAINED`` collection is simply an ordered list of other collections to search.  These can be nested arbitrarily (but cycles are of course not permitted), and may be redefined after creation.  A single-element ``CHAINED`` collection can thus be used as a sort of symbolic link, providing a stable name for a conceptual group of datasets whose definition may change over time.
+- A ``TAGGED`` collection is an explict list of individual datasets.
+- A ``CALIBRATION`` collection is an explict list of (dataset, validity range) tuples, where the validity range is a timespan.
 
-:tocdepth: 1
+A dataset is always in exactly one ``RUN`` collection, but may be in any number of ``TAGGED`` and/or ``CALIBRATION`` collections as well.
 
-.. Please do not modify tocdepth; will be fixed when a new Sphinx theme is shipped.
+It is also important to note that collections are always groups of *datasets*, not data IDs (again, see the `daf_butler documentation`_ for more information
+on these concepts).
+So, for example, a collection can contain *raws* or *coadds*, but not *visits* or *tracts*.
 
-.. sectnum::
+In addition, collections are constrained to hold only one dataset with a particular combination of dataset type (e.g. "calexp") and data ID, with the exception of ``CALIBRATION`` collections, for which there may only be one dataset with a particular dataset type and data ID at any point in time.
 
-.. TODO: Delete the note below before merging new content to the master branch.
+.. note::
+   Butler nomenclature in both Gen2 and Gen3 uses "dataset" to refer to entities that are of order one file (e.g. "a raw" or "a coadd" is "a dataset").
+   Informally, we also frequently refer to large sets of these datasets and/or their data IDs ("HSC PDR1" or "DESC DC2") as datasets, but I will avoid that usage whenever possible here.
+
+The naming patterns for collections proposed here are summarized in :ref:`table-overview`, with details and explanations in the following subsections.
+
+.. _table-overview:
+
+.. table:: Overview of collection naming conventions.
```

Comment:
This table is very useful, and I suspect we'll all be referring to it often to remember where things live for the first n-months of gen3 usage. My only complaint is that I had to read the rest of the document to figure out what some of the things in the table meant (e.g. the "unspecified" types), but I don't think there's any way around that. It would be nice to have links from the table name patterns to the relevant section of the doc, to make such lookups easier.

Reply:
Links are a good idea, but something I may defer until this content is transplanted into a long-term home, as links often don't survive those kinds of transitions well.